gobblin-modules/gobblin-compliance/src/main/java/gobblin/compliance/HivePartitionDataset.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package gobblin.compliance;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.metastore.api.FieldSchema;
+import org.apache.hadoop.hive.ql.metadata.Partition;
+
+import com.google.common.base.Optional;
+
+import gobblin.dataset.Dataset;
+
+
+/**
+ * A class to represent Hive Partition Dataset object used for gobblin-compliance
+ *
+ * @author adsharma
+ */
+public class HivePartitionDataset implements Dataset {
+  private Partition hivePartition;
```

Comment:
Half of the provided methods in HivePartitionDataset actually calls method on another object e.g. public String getTableName() {
    return this.hivePartition.getTable().getTableName();
  }

Second I want to restrict the number of methods called on HivePartitionDataset. Doing something like @Delegation private Partition hivePartition will defeat the purpose.